### PR TITLE
Statuslist: Merge issuer and verifier

### LINKED ISSUE #2850: Statuslist2021: add joint storage for verifier/issuer

### PR DESCRIPTION
**Pull request by gerardsn**
closes #2850 

This PR
- merges all issuer and verifier logic in 1 object
- rewrite verifiers
- adds storage to the verifier
- moves issuing of StatusList2021Credential from the issuer to the joint package
- new db table

Files will be renamed to the following: 
store.go -> issuer.go
credential_status.go -> verifier.go
Will have to wait until after this PR since Github saw the name changes as deleting/adding files.

### CODE REVIEW

File: vcr/statuslist2021/types.go [hunk 1]
```diff
@@ -49,14 +120,43 @@ type Entry struct {
...
+
+	// 'statusListCredential' must be a URL
+	if _, err := url.ParseRequestURI(e.StatusListCredential); err != nil {
+		return fmt.Errorf("parse StatusList2021Entry.statusListCredential URL: %w", err)
+	}
+
+	return nil
+}
+
 type CredentialSubject struct {
```
Comment (woutslakhorst) [inline vcr/statuslist2021/types.go:157]: godoc

File: vcr/statuslist2021/types.go [hunk 2]
```diff
@@ -35,6 +47,65 @@ const (
...
+// VerifySignFn verifies the signature on VC. The vcr.verifier injects its VerifySignature method here.
+type VerifySignFn func(credentialToVerify vc.VerifiableCredential, validateAt *time.Time) error
+
+// SignFn signs a VC according to the specified format. The vcr.issuer injects its signVC method here.
+type SignFn func(ctx context.Context, unsignedCredential vc.VerifiableCredential, credentialFormat string) (*vc.VerifiableCredential, error)
+
+var _ Issuer = (*CredentialStatus)(nil)
+var _ Verifier = (*CredentialStatus)(nil)
+
+type CredentialStatus struct {
```
Comment (woutslakhorst) [inline vcr/statuslist2021/types.go:97]: godoc

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
credential != credential in comments. Propose to use StatusList2021Credential where applicable.

 some minor things to fix in godoc.

Optionally: I find it a bit hard to read, what do you think about:
- naming the tables: `status_list`, `status_list_credential` and `status_list_entry` (or _revocation)?
- name the package `revocation` and the service struct `StatusList2021`?

**Comment by gerardsn** [comment]
I've added comments, and replaced credential with StatusList2021Credential where applicable.

Tables are renamed as proposed. (I think, status_list_credential and status_list are hard to keep apart)

I like revocation.StatusList2021 more than statuslist2021.CredentialStatus, but renaming the package also requires prefixing many of the exposed variables with StatusList2021. see https://github.com/nuts-foundation/nuts-node/pull/2851/commits/ca7a80ba57fd2ec1a76125b80106ebeef4103d8a. I'm not sure renaming the package is an improvement overall, but it would allow adding BitstringStatusList here. What do you think? (A better name for `CredentialStatus` would be good.) 

Yay or nay or removing the interfaces? They are not used...

File: storage/sql_migrations/006_statuslist.sql [hunk 3]
```diff
@@ -1,9 +1,28 @@
 -- migrate:up
--- status_list_credential: keeps track of all status list credentials issued by an issuer, and the highest status list index issued for each credential.
+
+-- status_list_credential: latest version of known credentials.
 create table status_list_credential
 (
     -- id: VC.credentialSubject.ID; URL where this status list credential can be downloaded.
-    subject_id                  varchar(500)    not null    primary key,
+    -- the credential served at this URL may change over time, so we only keep the latest
```
Comment (reinkrul) [inline storage/sql_migrations/006_statuslist.sql:7]: incorrect docs

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**